internal/incident/incidents.go

```diff
@@ -19,6 +20,89 @@ var (
 	currentIncidentsMu sync.Mutex
 )
 
+// LoadOpenIncidents loads all active (not yet closed) incidents from the database and restores all their states.
+// Returns error ony database failure.
+func LoadOpenIncidents(
+	ctx context.Context, db *icingadb.DB, logger *logging.Logger, runtimeConfig *config.RuntimeConfig, configFile *config.ConfigFile,
+) error {
+	var incidentRows []*IncidentRow
+	err := db.SelectContext(ctx, &incidentRows, db.BuildSelectStmt(&IncidentRow{}, &IncidentRow{})+` WHERE "recovered_at" IS NULL`)
+	if err != nil {
+		logger.Errorw("Failed to load active incidents from database", zap.Error(err))
+
+		return errors.New("failed to fetch open incidents")
+	}
+
+	incidents := make(map[*object.Object]*Incident)
+	for _, incidentRow := range incidentRows {
+		incident := &Incident{
+			db:               db,
+			runtimeConfig:    runtimeConfig,
+			StartedAt:        incidentRow.StartedAt.Time(),
+			incidentRowID:    incidentRow.ID,
+			configFile:       configFile,
+			SeverityBySource: map[int64]event.Severity{},
+			EscalationState:  map[escalationID]*EscalationState{},
+			Rules:            map[ruleID]struct{}{},
+			Recipients:       map[recipient.Key]*RecipientState{},
+		}
```

Comment:
You're duplicating the struct initialization here and it contains lots of "set to default values" (that primarily is initializing maps), so I think a `NewIncident()` function would help, otherwise changing anything becomes annoying (and if another required field is added, there's a good chance it might be forgotten in one place, which wouldn't happen if it was an argument of that `NewIncident()` function).